### DISCUSSION TITLE
Fedora: RPM update ends in erase of files

### DISCUSSION THREAD

**Discussion by Cyborgscode** [rustdesk/rustdesk#7469] [General]
### Bug Description

Once again, the RPM update removed the content instead of updating the system:

... I think i know what happens...

Check the order which post*install gets executed first and last .. it's the wrong order to do things!

[root@username ~]# dnf update https://github.com/rustdesk/rustdesk/releases/download/1.2.3-1/rustdesk-1.2.3-0.x86_64.rpm
Letzte Prüfung auf abgelaufene Metadaten: vor 2:39:44 am Do 21 Mär 2024 13:57:04 CET.
rustdesk-1.2.3-0.x86_64.rpm                                                             5.5 MB/s |  24 MB     00:04    
Abhängigkeiten sind aufgelöst.
========================================================================================================================
 Paket                       Architektur               Version                    Paketquelle                     Größe
========================================================================================================================
Aktualisieren:
 rustdesk                    x86_64                    1.2.3-0                    @commandline                     24 M

Transaktionszusammenfassung
========================================================================================================================
Aktualisieren  1 Paket

Gesamtgröße: 24 M
Ist dies in Ordnung? [j/N]: j
Pakete werden heruntergeladen:
Transaktionsüberprüfung wird ausgeführt
Transaktionsüberprüfung war erfolgreich.
Transaktion wird getestet
Transaktionstest war erfolgreich.
Transaktion wird ausgeführt
  Vorbereitung läuft    :                                                                                           1/1 
  Ausgeführtes Scriptlet: rustdesk-1.2.3-0.x86_64                                                                   1/2 
  Aktualisieren         : rustdesk-1.2.3-0.x86_64                                                                   1/2 
  Ausgeführtes Scriptlet: rustdesk-1.2.3-0.x86_64                                                                   1/2 
ln: die symbolische Verknüpfung '/usr/bin/rustdesk' konnte nicht angelegt werden: Die Datei existiert bereits

  Ausgeführtes Scriptlet: rustdesk-1.1.9-0.x86_64                                                                   2/2 
Removed "/etc/systemd/system/multi-user.target.wants/rustdesk.service".

  Aufräumen             : rustdesk-1.1.9-0.x86_64                                                                   2/2 
  Ausgeführtes Scriptlet: rustdesk-1.1.9-0.x86_64                                                                   2/2 
  Überprüfung läuft     : rustdesk-1.2.3-0.x86_64                                                                   1/2 
  Überprüfung läuft     : rustdesk-1.1.9-0.x86_64                                                                   2/2 

Aktualisiert:
  rustdesk-1.2.3-0.x86_64                                                                                               

Fertig!
[root@username ~]# systemctl status rustdesk.service --user 
Unit rustdesk.service could not be found.
[root@username ~]# systemctl status rustdesk.service
Unit rustdesk.service could not be found.
[root@username ~]# ls -la /usr/bin/rustdesk
ls: Zugriff auf '/usr/bin/rustdesk' nicht möglich: Datei oder Verzeichnis nicht gefunden
[root@username ~]# dnf reinstall https://github.com/rustdesk/rustdesk/releases/download/1.2.3-1/rustdesk-1.2.3-0.x86_64.rpm
Letzte Prüfung auf abgelaufene Metadaten: vor 2:42:10 am Do 21 Mär 2024 13:57:04 CET.
rustdesk-1.2.3-0.x86_64.rpm                                                             5.8 MB/s |  24 MB     00:04    
Abhängigkeiten sind aufgelöst.
========================================================================================================================
 Paket                       Architektur               Version                    Paketquelle                     Größe
========================================================================================================================
Neuinstallieren:
 rustdesk                    x86_64                    1.2.3-0                    @commandline                     24 M

Transaktionszusammenfassung
========================================================================================================================

Gesamtgröße: 24 M
Installationsgröße: 62 M
Ist dies in Ordnung? [j/N]: j
Pakete werden heruntergeladen:
Transaktionsüberprüfung wird ausgeführt
Transaktionsüberprüfung war erfolgreich.
Transaktion wird getestet
Transaktionstest war erfolgreich.
Transaktion wird ausgeführt
  Vorbereitung läuft    :                                                                                           1/1 
  Ausgeführtes Scriptlet: rustdesk-1.2.3-0.x86_64                                                                   1/2 
Failed to stop rustdesk.service: Unit rustdesk.service not loaded.

  Neuinstallieren       : rustdesk-1.2.3-0.x86_64                                                                   1/2 
  Ausgeführtes Scriptlet: rustdesk-1.2.3-0.x86_64                                                                   1/2 
Created symlink /etc/systemd/system/multi-user.target.wants/rustdesk.service → /etc/systemd/system/rustdesk.service.

  Ausgeführtes Scriptlet: rustdesk-1.2.3-0.x86_64                                                                   2/2 
  Aufräumen             : rustdesk-1.2.3-0.x86_64                                                                   2/2 
  Ausgeführtes Scriptlet: rustdesk-1.2.3-0.x86_64                                                                   2/2 
  Überprüfung läuft     : rustdesk-1.2.3-0.x86_64                                                                   1/2 
  Überprüfung läuft     : rustdesk-1.2.3-0.x86_64                                                                   2/2 

Erneut installiert:
  rustdesk-1.2.3-0.x86_64                                                                                               

Fertig!
[root@username ~]# systemctl status rustdesk.service
● rustdesk.service - RustDesk
     Loaded: loaded (/etc/systemd/system/rustdesk.service; enabled; preset: disabled)
    Drop-In: /usr/lib/systemd/system/service.d
             └─10-timeout-abort.conf
     Active: active (running) since Thu 2024-03-21 16:39:23 CET; 4s ago
   Main PID: 212193 (rustdesk)
      Tasks: 7 (limit: 9343)
     Memory: 4.6M
        CPU: 1.566s
     CGroup: /system.slice/rustdesk.service
             ├─212193 /usr/bin/rustdesk --service
             └─212605 sh -c "ps -u 1000 -f | grep -E 'Xwayland' | grep -v 'grep' | tail -1 | awk '{print \$2}' | xargs >

Mär 21 16:39:23 username.>ERASED< systemd[1]: Started rustdesk.service - RustDesk.

### How to Reproduce

update an older rustdesk with dnf.

### Expected Behavior

see description

### Operating system(s) on local side and remote side

FEDORA LINUX 38/39

### RustDesk Version(s) on local side and remote side

1.1.x -> 1.2.0

### Screenshots

see above log.

**Comment by xlionjuan** [comment]
You should use 1.2.3 or 1.2.4 nightly
